compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5504,7 +5505,11 @@ public void visit(BLangFromClause fromClause) {
     @Override
     public void visit(BLangJoinClause joinClause) {
         boolean prevBreakEnv = this.breakToParallelQueryEnv;
-        this.breakToParallelQueryEnv = true;
+        if (joinClause.collection.getKind() == NodeKind.QUERY_EXPR ||
+                (joinClause.collection.getKind() == NodeKind.GROUP_EXPR
+                        && ((BLangGroupExpr) joinClause.collection).expression.getKind() == NodeKind.QUERY_EXPR)) {
+            this.breakToParallelQueryEnv = true;
```

Comment:
Shall we explain what is parallel query expression breaking near this variables definition with a comment.